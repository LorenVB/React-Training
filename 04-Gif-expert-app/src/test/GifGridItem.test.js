const { shallow } = require("enzyme");
const { default: toJson } = require("enzyme-to-json");
const { GifGridItem } = require("../components/GifGridItem")


describe('should match snapshot', () => {
    
    const title= 'Un titulo';
    const url = 'https://localhost/smth.es'
    const wrapper= shallow(<GifGridItem title={title} url={url}/>);

    it('renders correctly', ()=> {
        
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('should have one paragraph with the title', ()=>{

        const p= wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    it('should have similar image to the props url and alt',()=>{
        const img= wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    it('should have the animated fadeIn css class', ()=>{
        const div= wrapper.find('div');
        const className= div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);
    })
});
