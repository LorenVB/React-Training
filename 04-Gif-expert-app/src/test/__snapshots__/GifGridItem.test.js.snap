// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot renders correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/smth.es"
  />
  <p>
    Un titulo
  </p>
</div>
`;
